src/cdk-experimental/menu/menu-item.spec.ts

```diff
@@ -61,27 +61,59 @@ describe('MenuItem', () => {
   });
 
   describe('with complex inner elements', () => {
-    let fixture: ComponentFixture<ComplexMenuItem>;
     let menuItem: CdkMenuItem;
 
-    beforeEach(async(() => {
+    /**
+     * Build a component for testing and render it.
+     * @param componentClass the component to create
+     */
+    function createComponent<T>(componentClass: Type<T>) {
+      let fixture: ComponentFixture<T>;
+
       TestBed.configureTestingModule({
         imports: [CdkMenuModule],
-        declarations: [ComplexMenuItem],
+        declarations: [componentClass, MatIcon],
         providers: [{provide: CDK_MENU, useClass: CdkMenu}],
       }).compileComponents();
-    }));
 
-    beforeEach(() => {
-      fixture = TestBed.createComponent(ComplexMenuItem);
+      fixture = TestBed.createComponent(componentClass);
       fixture.detectChanges();
 
       menuItem = fixture.debugElement.query(By.directive(CdkMenuItem)).injector.get(CdkMenuItem);
```

Comment:
Nit: consider having the function return this instead of assigning it to an outside variable?
- Pro: might make tests a little more readable
- Con: slightly more boilerplate

Reply:
Do you mean having a function which returns the menu item rather than setting the variable? 

Setting a variable seems to be more consistent with the rest of the tests. Thoughts?